dir.c

```diff
@@ -2226,8 +2224,13 @@ glob_opendir(ruby_glob_entries_t *ent, DIR *dirp, int flags, rb_encoding *enc)
         }
         ruby_qsort(ent->sort.entries, ent->sort.count, sizeof(ent->sort.entries[0]),
                    glob_sort_cmp, NULL);
+        return ent;
+
+      nomem:
```

Comment:
Just out of curiosity, why not the end of the function?

Reply:
I thought this is the right place to put `nomem:` because all `goto nomem;` happens inside of this scope.

But the end of the function also seems perfect.  No strong opinion now.

Reply:
https://github.com/ruby/ruby/pull/3247/commits/12a56d74cad9ae2158a1485023ecee590ac67a73